File: 2610-Convert_an_Array_Into_2D_Array_With_Conditions/Convert_an_Array_Into_2D_Array_With_Conditions.java
import java.util.*;

class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        int[] freq = new int[nums.length + 1];
        
        List<List<Integer>> ans = new ArrayList<>();
        for(int i: nums){
            if(freq[i] >= ans.size()) {
                ans.add(new ArrayList<>());
            }
            ans.get(freq[i]).add(i);
            freq[i]++;
        }
        
        return ans;
    }
}

File: 2610-ConvertAnArrayInto2DArrayWithConditions/ConvertAnArrayInto2DArrayWithConditions.go
package convertanarrayinto2darraywithconditions

func findMatrix(nums []int) [][]int {
	freq := make(map[int]int)
	ans := make([][]int, 0)

	for _, val := range nums {
		if freq[val] >= len(ans) {
			ans = append(ans, []int{})
		}

		ans[freq[val]] = append(ans[freq[val]], val)
		freq[val]++
	}

	return ans
}
